### PR TITLE
Initial project setup

### PR DESCRIPTION
**Pull request by AlexStajosRack**
Initial project setup with the following goals:
- Multi-service compatible monorepo
- Node compatible
- CircleCI
- User, Dev and Prod stages
- Linting
- Initial test of running Puppeteer in a lambda using Layers

Not included: Testing

### CODE REVIEW

File: services/identity-source/main.tf [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      "logs:*",
+      "xray:PutTraceSegments",
+      "xray:PutTelemetryRecords",
+    ]
+
+    resources = ["*"]
+  }
+}
+
+# Lambda Layers
```
Comment (ryandub) [inline services/identity-source/main.tf]: 👍

File: services/identity-source/main.tf [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  function_name                  = "${var.stage}_${var.service_name}-add-identity-source"
+  filename                       = "${data.archive_file.add_identity_source_archive.output_path}"
+  role                           = "${aws_iam_role.add_identity_source_execution_lambda_role.arn}"
+  handler                        = "index.addIdentitySource"
+  memory_size                    = 1600
+  timeout                        = 600
+  source_code_hash               = "${base64sha256(file("${data.archive_file.add_identity_source_archive.output_path}"))}"
+  runtime                        = "nodejs10.x"
+  description                    = "Web automation to add custom identity source. (stage: ${var.stage})"
+  layers                         = ["${aws_lambda_layer_version.chrome_aws_lambda_layer.arn}"]
```
Comment (ryandub) [inline services/identity-source/main.tf]: I think that the Xplat terraform module can support layers, it just needs to have it added to https://github.com/rackerlabs/xplat-terraform-modules/blob/master/modules/lambda/main.tf#L47 and https://github.com/rackerlabs/xplat-terraform-modules/blob/master/modules/lambda/variables.tf

File: services/identity-source/main.tf [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  role   = "${aws_iam_role.add_identity_source_execution_lambda_role.id}"
+  policy = "${data.aws_iam_policy_document.base_lambda_policy.json}"
+}
+
+resource "aws_lambda_function" "lambda" {
+  function_name                  = "${var.stage}_${var.service_name}-add-identity-source"
+  filename                       = "${data.archive_file.add_identity_source_archive.output_path}"
+  role                           = "${aws_iam_role.add_identity_source_execution_lambda_role.arn}"
+  handler                        = "index.addIdentitySource"
+  memory_size                    = 1600
```
Comment (ryandub) [inline services/identity-source/main.tf]: That's a lot of memory - is it needed?

Comment (AlexStajosRack) [reply services/identity-source/main.tf]: I'm not sure. Basing it on this line from the chrome layer recommendaiton: 
"You should allocate at least 512 MB of RAM to your Lambda, however 1600 MB (or more) is recommended."

I can drop it to 512 for now.
https://github.com/alixaxel/chrome-aws-lambda

Comment (ryandub) [reply services/identity-source/main.tf]: ah, ok. Just sanity checking. If it is recommended then it's fine to leave. Thanks for explaining!

**Pull request merged**